### PR TITLE
Continue Reading Button Update

### LINKED ISSUE #19: Make "continue reading" button bigger, like in tachiyomi 0.14.7
### Describe your suggested feature

Mihon v0.16.0 button size:
Screenshot_20240116-104546_Mihon

Tachiyomi 0.14.7 button size:
Screenshot_20240116-092943_Tachiyomi

### Other details

Making size configurable would be ideal.

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a short but informative title.
- [X] If this is an issue with an official extension, I should be opening an issue in the [extensions repository](https://github.com/tachiyomiorg/extensions/issues/new/choose).
- [X] I have updated the app to version **[0.15.3](https://github.com/mihonapp/mihon/releases/latest)**.
- [X] I will fill out all of the requested information in this form.

### PR DESCRIPTION
**Pull request by Animeboynz**
Allows the Continue Reading Button to have different sized for each Display Modes to fix #19.
I have done this specifically so that the Continue Reading buttons are larger, but not on the **Compact** display mode so that it does not cover too much of the title.

Testes on Phone and Tablet, with different Grid Sizes

### Images
| Image 1: OLD | Image 2: NEW |
| ------- | ------- |
| Tachiyomi_OLD | Tachiyomi_NEW |

### CODE REVIEW

File: app/src/main/java/eu/kanade/presentation/library/components/CommonMangaItem.kt [hunk 1]
```diff
@@ -50,7 +50,13 @@ object CommonMangaItemDefaults {
     const val BrowseFavoriteCoverAlpha = 0.34f
 }
 
-private val ContinueReadingButtonSize = 28.dp
+private val ContinueReadingButtonSize = mapOf(
+    "Compact" to 28.dp,
+    "Comfortable" to 35.dp,
+    "CoverOnly" to 35.dp,
+    "List" to 28.dp
+)
```
Comment (AntsyLich) [inline app/src/main/java/eu/kanade/presentation/library/components/CommonMangaItem.kt:58]: make it 2 variables and use them instead of this jank

Comment (AntsyLich) [reply app/src/main/java/eu/kanade/presentation/library/components/CommonMangaItem.kt:58]: also make it 32.dp

Comment (Animeboynz) [reply app/src/main/java/eu/kanade/presentation/library/components/CommonMangaItem.kt]: Completed this now.

**Comment by AntsyLich** [comment]
Add a size parameter to `ContinueReadingButton` pass it to the original `Modifier` like before. and remove all the comments.

**Comment by Animeboynz** [comment]
Not too used to Kotlin so I'm unsure about a few things.
If I add the Size Modifier back to the ContinueReadingButton()
```@Composable
private fun ContinueReadingButton(
    modifier: Modifier = Modifier,
    onClickContinueReading: () -> Unit,
) {
    Box(modifier = modifier) {
        FilledIconButton(
            onClick = onClickContinueReading,
            shape = MaterialTheme.shapes.small,
            modifier = Modifier.size(ContinueReadingButtonSize),
            colors = IconButtonDefaults.filledIconButtonColors(
                containerColor = MaterialTheme.colorScheme.primaryContainer.copy(alpha = 0.9f),
                contentColor = contentColorFor(MaterialTheme.colorScheme.primaryContainer),
            ),
        ) {
            Icon(
                imageVector = Icons.Filled.PlayArrow,
                contentDescription = stringResource(MR.strings.action_resume),
                modifier = Modifier.size(16.dp),
            )
        }
    }
}
```

Even when I pass another value through. It overrides it.
```@Composable
fun MangaComfortableGridItem(
    coverData: tachiyomi.domain.manga.model.MangaCover,
    title: String,
    onClick: () -> Unit,
    onLongClick: () -> Unit,
    isSelected: Boolean = false,
    titleMaxLines: Int = 2,
    coverAlpha: Float = 1f,
    coverBadgeStart: (@Composable RowScope.() -> Unit)? = null,
    coverBadgeEnd: (@Composable RowScope.() -> Unit)? = null,
    onClickContinueReading: (() -> Unit)? = null,
) {
    GridItemSelectable(
        isSelected = isSelected,
        onClick = onClick,
        onLongClick = onLongClick,
    ) {
        Column {
            MangaGridCover(
                cover = {
                    MangaCover.Book(
                        modifier = Modifier
                            .fillMaxWidth()
                            .alpha(if (isSelected) GridSelectedCoverAlpha else coverAlpha),
                        data = coverData,
                    )
                },
                badgesStart = coverBadgeStart,
                badgesEnd = coverBadgeEnd,
                content = {
                    if (onClickContinueReading != null) {
                        ContinueReadingButton(
                            modifier = Modifier
                                .padding(ContinueReadingButtonGridPadding)
                                .align(Alignment.BottomEnd)
                                .size(ContinueReadingButtonSizeLarge),
                            onClickContinueReading = onClickContinueReading,
                        )
                    }
                },
            )
            GridItemTitle(
                modifier = Modifier.padding(4.dp),
                title = title,
                style = MaterialTheme.typography.titleSmall,
                minLines = 2,
                maxLines = titleMaxLines,
            )
        }
    }
}```

**Comment by AntsyLich** [comment]
Merged as part of e17f70f7226ea031fc1f962c9dfea3e404ba53ad

**Pull request closed without merging**